.github/workflows/ci.yml

```
@@ -0,0 +1,29 @@
+# This workflow will do a clean install of node dependencies, build the source code and run tests across different versions of node
+# For more information see: https://help.github.com/actions/language-and-framework-guides/using-nodejs-with-github-actions
+
+name: Node.js CI
+
+on:
+  push:
+    branches: [ main ]
```

```suggestion
    branches-ignore: [ main ]
```

I recommend running on push to each branch, and then again on PR, but not on pushes to `main`. As it is, you won't know that a branch is failing tests until you do the PR, which is potentially wasting time for the committer and reviewer.